### PR TITLE
feat: migrate less to token for Anchor

### PR DESCRIPTION
**Pull request by MuxinFeng**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/41884

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     migrate less to token for Anchor      |
| 🇨🇳 Chinese |    迁移 `Anchor` 组件的 less token       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 23d7873</samp>

Added a new interface to export anchor padding tokens and updated the documentation for the anchor component variables. This allows users to customize the anchor component styles more easily and migrate from the deprecated Less variables to the new Component Tokens.

### <samp>🤖 Generated by Copilot at 23d7873</samp>

*  Add `ComponentToken` interface to export common anchor padding properties as tokens for customization and update `AnchorToken` interface to extend it ([link](https://github.com/ant-design/ant-design/pull/42141/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL6-R13))
*  Update `genComponentStyleHook` function to accept a third argument that returns the `ComponentToken` values based on the `FullToken` and move the anchor padding properties from the `anchorToken` to the third argument function ([link](https://github.com/ant-design/ant-design/pull/42141/files?diff=unified&w=0#diff-317d329fe73e0dd37dc2416f47e9ddb7f377ece859e71f5f36a65025e7f8581eL161-R179))
*  Add documentation for the anchor component variables in English and Chinese versions, showing the mapping between the deprecated Less variables and the new Component Tokens, and providing some notes on the style changes ([link](https://github.com/ant-design/ant-design/pull/42141/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL16-R26), [link](https://github.com/ant-design/ant-design/pull/42141/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L16-R26))

### CODE REVIEW

File: docs/react/migrate-less-variables.zh-CN.md [hunk 1]
```diff
@@ -13,7 +13,16 @@ title: Less 变量迁移 Design Token
 
 <!-- ### Alert 警告提示 -->
 
-<!-- ### Anchor 锚点 -->
+## Anchor 锚点
+
+<!-- prettier-ignore -->
+| Less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@anchor-bg` | - | 由于样式变化已废弃 |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:21]: 这个看了下应该还是有的，可以加上

File: docs/react/migrate-less-variables.zh-CN.md [hunk 2]
```diff
@@ -13,7 +13,16 @@ title: Less 变量迁移 Design Token
...
+| `@anchor-border-color` | - | 由于样式变化已废弃 |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:22]:
```suggestion
| `@anchor-border-color` | `colorSplit` | - |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 3]
```diff
@@ -13,7 +13,16 @@ title: Less 变量迁移 Design Token
...
+| `@anchor-link-top` | `anchorPaddingBlock` | - |
+| `@anchor-link-left` | `anchorPaddingInline` | - |
+| `@anchor-link-padding` | - | 由于样式变化已废弃 |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:25]: 这个也应该还是有的，只是被拆开写了

File: docs/react/migrate-less-variables.zh-CN.md (hunk 1 above)
Comment (MuxinFeng) [reply docs/react/migrate-less-variables.zh-CN.md:21]: 嗯嗯，好嘞，我看他写死在代码里了，就不太确定要不要改

File: docs/react/migrate-less-variables.zh-CN.md (hunk 2 above)
Comment (MuxinFeng) [reply docs/react/migrate-less-variables.zh-CN.md:22]: 这里还有点疑惑，因为找不到用 `@anchor-border-color` 的地方了

File: docs/react/migrate-less-variables.zh-CN.md [hunk 4]
```diff
@@ -13,7 +13,16 @@ title: Less 变量迁移 Design Token
 
 <!-- ### Alert 警告提示 -->
 
-<!-- ### Anchor 锚点 -->
+## Anchor 锚点
+
+<!-- prettier-ignore -->
+| Less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@anchor-bg` | `background` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:21]: 这个可以废弃，备注“可以由 `className` 或 `style` 直接修改”

File: docs/react/migrate-less-variables.zh-CN.md [hunk 5]
```diff
@@ -13,7 +13,16 @@ title: Less 变量迁移 Design Token
...
+| `@anchor-border-color` | `colorSplit` | 全局 Token |
+| `@anchor-link-top` | `paddingBlock` | - |
+| `@anchor-link-left` | `paddingInline` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:24]: 这两个应该废弃掉了，v4 都没有在用

File: docs/react/migrate-less-variables.zh-CN.md [hunk 6]
```diff
@@ -13,7 +13,16 @@ title: Less 变量迁移 Design Token
...
+| `@anchor-link-padding` | - | 由于样式变化已废弃 |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:25]: 这个备注一下使用上面两个 token 就可以

File: docs/react/migrate-less-variables.zh-CN.md (hunk 5 above)
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:24]:
```suggestion
| `@anchor-link-top` | `linkPaddingBlock` | - |
| `@anchor-link-left` | `linkPaddingInlineStart` | - |
```

**Review by MadCcc — APPROVED** [review]

**Pull request merged**